src/main/java/com/thealgorithms/sorts/SmoothSort.java

```diff
@@ -0,0 +1,227 @@
+package com.thealgorithms.sorts;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+
+/**
+ * Wikipedia: https://en.wikipedia.org/wiki/Smoothsort
+ */
+public final class SmoothSort implements SortAlgorithm {
+
+    public SmoothSort() {
+    }
+
+    private static Integer[] getLeonardoNumbers() {
+        Integer[] leonardoNumbers = {1, 1, 3, 5, 9, 15, 25, 41, 67, 109, 177, 287, 465, 753, 1219, 1973, 3193, 5167, 8361, 13529, 21891, 35421, 57313, 92735, 150049, 242785, 392835, 635621, 1028457, 1664079, 2692537, 4356617, 7049155, 1405773, 18454929, 29860703, 48315633, 78176337, 126491971,
+            204668309, 331160281, 535828591};
```

Comment:
Why not to use [`LeonardoNumber`](https://github.com/TheAlgorithms/Java/blob/a710fe11c47bdfb80342f851c189d167035b7b1c/src/main/java/com/thealgorithms/maths/LeonardoNumber.java) here?

Reply:
The [LeonardoNumber](https://github.com/TheAlgorithms/Java/blob/a710fe11c47bdfb80342f851c189d167035b7b1c/src/main/java/com/thealgorithms/maths/LeonardoNumber.java) class is recursively calculating the Leonardo Numbers. I did not want to calculate(or re-calculate) the numbers everytime. I shall modify my code and use [LeonardoNumber](https://github.com/TheAlgorithms/Java/blob/a710fe11c47bdfb80342f851c189d167035b7b1c/src/main/java/com/thealgorithms/maths/LeonardoNumber.java) to store the results in an array before I start creating the heap and use the results stored in that array when needed.

Reply:
@vil02 I would however think recrsivly calculating these numbers ( before or during the execution ) might not be a great idea and would just increase the execution time as it will run the same recursion stacks multiple times. If you still suggest using LeonardoNumber.java I shall go ahead and make changes as mentioned above.

Reply:
@JAPNITSINGH we can define this array as `static`. For time being, I do not have a clear preference. The option for the future, is definitely to provide a way to compute Leonardo numbers more efficiently.